src/sentry/api/endpoints/project_ownership.py

```diff
@@ -28,17 +30,16 @@ def validate_raw(self, attrs, source):
 
         schema = dump_schema(rules)
 
+        owners = {o for rule in rules for o in rule.owners}
+        actors = resolve_actors(owners, self.context['ownership'].project_id)
+
         bad_actors = []
-        for rule in rules:
-            for owner in rule.owners:
-                try:
-                    resolve_actor(owner, self.context['ownership'].project_id)
-                except UnknownActor:
-                    if owner.type == 'user':
-                        bad_actors.append(owner.identifier)
-
-                    if owner.type == 'team':
-                        bad_actors.append(u'#{}'.format(owner.identifier))
+        for owner, actor in six.iteritems(actors):
+            if actor is None:
+                if owner.type == 'user':
+                    bad_actors.append(owner.identifier)
+                elif owner.type == 'team':
+                    bad_actors.append(u'#{}'.format(owner.identifier))
```

Comment:
Seems like maybe this formatting method should be on `Owner.__unicode__` or something?

Reply:
Seems reasonable. I'll do that.